### PR TITLE
Added daemonset, new fluentd plugin, less permissive rbac

### PR DESCRIPTION
**Pull request by chaitanyaphalak**

### CODE REVIEW

File: helm-chart/splunk-kubernetes-metrics/templates/clusterRoleBinding.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ metadata:
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: ClusterRole
-  name: "system:heapster"
+  name: "system:nodes"
```
Comment (chaitanyaphalak) [inline helm-chart/splunk-kubernetes-metrics/templates/clusterRoleBinding.yaml]: I think using https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/rbac/kubelet-api-auth/kubelet-api-admin-role.yaml might be better.

File: helm-chart/splunk-kubernetes-metrics/values.yaml [hunk 2]
```diff
@@ -74,25 +74,17 @@ splunk:
 
 
 image:
-  heapsterTag: v1.5.1
-  sidecarTag: 1.0.1
+  name: chaitanyaphalak/k8s-new-metrics:beta
```
Comment (dtregonning) [inline helm-chart/splunk-kubernetes-metrics/values.yaml]: move to a more official repo

File: helm-chart/splunk-kubernetes-metrics/values.yaml [hunk 3]
```diff
@@ -74,25 +74,17 @@ splunk:
...
   pullPolicy: IfNotPresent
 
 
 resources:
-  heapster:
+  fluent:
     limits:
-      cpu: 100m
+      cpu: 200m
```
Comment (dtregonning) [inline helm-chart/splunk-kubernetes-metrics/values.yaml]: why the change?

File: helm-chart/splunk-kubernetes-metrics/templates/deployment.yaml [hunk 4]
```diff
@@ -39,50 +28,22 @@ spec:
...
-                  name: {{ template "splunk-kubernetes-metrics.fullname" . }}
-                  key: splunk_hec_token
-          resources:
-{{ toYaml .Values.resources.sidecar | indent 12 }}
-          volumeMounts:
+      - name: fluentd
+        image: {{ .Values.image.name }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        env:
+          - name: MY_NODE_NAME
```
Comment (dtregonning) [inline helm-chart/splunk-kubernetes-metrics/templates/deployment.yaml]: should we change this from SPLUNK_HEC_HOST?

File: helm-chart/splunk-kubernetes-metrics/templates/deployment.yaml [hunk 5]
```diff
@@ -1,29 +1,18 @@
-apiVersion: apps/v1beta1
-kind: Deployment
+apiVersion: extensions/v1beta1
```
Comment (dtregonning) [inline helm-chart/splunk-kubernetes-metrics/templates/deployment.yaml]: I'd leave as apps as extensions is being deprecated at some point - https://github.com/kubernetes/kubernetes/issues/43214

File: helm-chart/splunk-kubernetes-metrics/values.yaml (hunk 2 above)
Comment (chaitanyaphalak) [reply helm-chart/splunk-kubernetes-metrics/values.yaml]: need to figure out how to to push to splunk dockerhub

Comment (dtregonning) [reply helm-chart/splunk-kubernetes-metrics/values.yaml]: ok leave for now

**Comment by dtregonning** [comment]
@chaitanyaphalak - please resubmit against develop
